include/evmc/evmc.h

```diff
@@ -275,6 +275,12 @@ enum evmc_status_code
      */
     EVMC_CONTRACT_VALIDATION_FAILURE = 13,
 
+    /**
+     * A trap instruction has been hit during exection.
+     * This is useful for WebAssembly implementations, where the `unreachable`
+     * instruction and various execution errors are turned into a trap.
+     */
+    EVMC_TRAP_INSTRUCTION = 14,
```

Comment:
Should this be merged with `EVMC_INVALID_INSTRUCTION`?

Reply:
Rather not, because trap can be caused not only by a contract design.

Reply:
Decision: use `EVMC_UNREACHABLE_INSTRUCTION` here.

Reply:
In addition we could have `EVMC_WEBASSEMBLY_TRAP` for the generic trap case as specified by the wasm spec.